src/server/task/server/api_server.go

```diff
@@ -0,0 +1,69 @@
+package server
+
+import (
+	"path"
+	"strings"
+
+	col "github.com/pachyderm/pachyderm/v2/src/internal/collection"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	"github.com/pachyderm/pachyderm/v2/src/taskapi"
+
+	etcd "go.etcd.io/etcd/client/v3"
+)
+
+type apiServer struct {
+	ppsPrefix, pfsPrefix string
+	etcdClient           *etcd.Client
+}
+
+func NewAPIServer(base, pps, pfs string, etcdClient *etcd.Client) taskapi.APIServer {
+	return &apiServer{
+		ppsPrefix:  path.Join(base, pps),
+		pfsPrefix:  path.Join(base, pfs),
+		etcdClient: etcdClient,
+	}
+}
+
+func translateTaskState(state task.State) taskapi.State {
+	switch state {
+	case task.State_RUNNING:
+		return taskapi.State_RUNNING
+	case task.State_SUCCESS:
+		return taskapi.State_SUCCESS
+	case task.State_FAILURE:
+		return taskapi.State_FAILURE
+	}
+	return taskapi.State_STATE_UNKNOWN
+}
+
+func (a *apiServer) ListTask(req *taskapi.ListTaskRequest, server taskapi.API_ListTaskServer) (retError error) {
+	ctx := server.Context()
+	if req.Namespace == "" {
+		return errors.New("Must include a task namespace")
+	}
+	prefix := a.ppsPrefix
+	if strings.HasPrefix(req.Namespace, "storage") {
+		prefix = a.pfsPrefix
+	}
+	etcdCols := task.NewNamespaceEtcd(a.etcdClient, prefix, req.Namespace)
+	var taskData task.Task
+	etcdCols.TaskCol.ReadOnly(ctx).List(&taskData, col.DefaultOptions(), func(key string) error {
```

Comment:
Looks like we are dropping the list error, we can just return the result of this call.

Reply:
Also, this functionality should probably be built into the task service. This is pulling the details out of a particular implementation. I think we could handle all of this (including the claim state) internally and expose a list method through the service interface. We would still want to have a separate type that we convert to though.

Reply:
reworked to put everything in the task service, with simple wrapper implementations